compiler/ghc/default.nix

```diff
@@ -351,7 +388,40 @@ stdenv.mkDerivation (rec {
     # Our Cabal compiler name
     haskellCompilerName = "ghc-${version}";
 
-    configured-src = configured-src;
+    # This uses a similar trick to `pkgs.srcOnly` to get the configured src
+    # We could add `configured-src` as an output of the ghc derivation, but
+    # having it as its own derivation means it can be accessed quickly without
+    # building GHC.
+    configured-src = stdenv.mkDerivation ({
+      name = name + "-configured-src";
+      inherit
+        buildInputs
+        version
+        nativeBuildInputs
+        patches
+        src
+        strictDeps
+        depsBuildTarget
+        depsTargetTarget
+        depsTargetTargetPropagated
+        postPatch
+        preConfigure
+        configurePlatforms
+        configureFlags
+        outputs
+        ;
+
+      # Including all the outputs (not just $out) causes `mkDerivation` to use the nixpkgs multiple-outputs.sh hook.
+      # This hook changes the arguments passed to `configure`.
+      installPhase = ''
+        cp -r . $out
+        mkdir $doc
+        mkdir $generated
+      '';
+      phases = [ "unpackPhase" "patchPhase" ]
+            ++ lib.optional (ghc-patches != []) "autoreconfPhase"
+            ++ [ "configurePhase" "installPhase"];
```

Comment:
So we are technically re-creating the configured source; but it's not the identical one we built GHC from.
I hope this holds in all situations.

Reply:
If it does become an issue we can add `configured-src` to `outputs` and copy the source there in `postConfigure`.  That has the drawback of requiring the compiler to be built to view the `configured-src`.